aten/src/THCUNN/generic/Col2Im.cu

```diff
@@ -0,0 +1,113 @@
+#ifndef THC_GENERIC_FILE
+#define THC_GENERIC_FILE "generic/Col2Im.cu"
+#else
+
+static inline void THNN_(Col2Im_shapeCheck)(
+                         THCState *state,
+                         THCTensor *input,
+                         THCTensor *gradOutput,
+                         int outputHeight, int outputWidth,
+                         int kH, int kW, int dH, int dW,
+                         int padH, int padW, int sH, int sW) {
+
+  THArgCheck(kW > 0 && kH > 0, 6,
+             "kernel size should be greater than zero, but got kH: %d kW: %d", kH, kW);
+  THArgCheck(sW > 0 && sH > 0, 12,
+             "stride should be greater than zero, but got sH: %d sW: %d", sH, sW);
+  THArgCheck(dW > 0 && dH > 0, 8,
+             "dilation should be greater than zero, but got dH: %d dW: %d", dH, dW);
+
+  int ndim = THCTensor_(nDimension)(state, input);
+  THCUNN_argCheck(state, ndim == 2 || ndim == 3, 2, input,
+                  "2D or 3D input tensor expected but got %s");
+
+  int batch_dim = (ndim == 4) ? 0 : -1;
+  long nInputPlane  = input->size[batch_dim + 1];
+  long inputLength  = input->size[batch_dim + 2];
+
+  long nOutputPlane = nInputPlane / (kW * kH);
+
+  if (outputWidth < 1 || outputHeight < 1) {
+    THError("Given input size: (%d x %d). "
+            "Calculated output size: (%d x %d x %d). Output size is too small",
+            nInputPlane, inputLength, nOutputPlane, outputHeight, outputWidth);
```

Comment:
Some of these arguments are `long`. On a 64-bit machine (we only support pytorch for 64-bit), `long` is 64-bit, so the descriptor `%d` doesn't fit those.

You can use `%lld` and cast things like `nOutputPlane` to `(long long)nOutputPlane` to get that to work.